client/src/lib/dependency-injection/inject.js

```diff
@@ -24,34 +31,21 @@ const inject = (dependencies, mapDependenciesToProps, getContext) => (Component)
       context to use throughout the component.
     `);
   }
+
   // eslint-disable-next-line react/prefer-stateless-function
   class Injected extends React.Component {
     render() {
       let props = {};
       const context = getContext ? getContext(this.props) : null;
-      const deps = dependencies;
-      if (deps) {
-        if (!Array.isArray(deps)) {
-          throw new Error(`
-          withInjector() passed an argument for dependencies that is ${typeof deps}. 
-          Must be a string or array of named dependencies.
-        `);
-        }
-        const resolved = deps.map(dep => this.context.injector.get(dep, context));
-        if (mapDependenciesToProps) {
-          props = mapDependenciesToProps(...resolved);
-          if (typeof props !== 'object') {
-            throw new Error(`
-              mapDepedenciesToProps parameter passed to inject() 
-              should return an object that maps prop names to dependencies
-             `);
-          }
-        } else {
+      if (dependencies) {
+        const resolved = dependencies.map(dep => this.context.injector.get(dep, context));
+
+        props = (mapDependenciesToProps)
+          ? mapDependenciesToProps(...resolved)
```

Comment:
oh true, I may have left it out and confused it with the `dependencies` check which I shifted above, will add it back in